appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaFormEditor.java

```diff
@@ -187,6 +187,10 @@ String getFileContent() {
     setSize("100%", "100%");
   }
 
+  public boolean shouldDisplayHiddenComponents() {
```

Comment:
See my other note about a getter for the state of the checkbox without exposing the underlying object model.